walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,90 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // throw new UnsupportedOperationException("TODO: Implement this method."); 
+    long reqDuration = request.getDuration();
+    Collection<String> reqAttendees = request.getAttendees();
+    ArrayList<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+
+    // Special cases
+    if (reqDuration > TimeRange.WHOLE_DAY.duration()) {
+        return possibleTimes;
+    }
+
+    // Collection of clashing events' (events where at least one attendee belongs to reqAttendees) timeranges
+    ArrayList<TimeRange> clashingEventTimes = new ArrayList<TimeRange>();
+
+    // Check each event in events collection
+    for (Event event : events) {
```

Comment:
This loop would be a great place to use the Java Streams library - it will shorten your code a lot!

clashingEventTimes = events.stream().filter(event -> isClashingEvent(event)).toList();

Reply:
I tried to implement it using streams but I can't figure out why it gives me this error... Do you have any idea why?
![Screenshot_2](https://user-images.githubusercontent.com/50647023/85698531-5662e100-b70d-11ea-9dfe-daf48eb7b2c2.jpg)
![Screenshot_1](https://user-images.githubusercontent.com/50647023/85698523-54991d80-b70d-11ea-932f-e10b38dc05a7.jpg)

Reply:
Yep you’ll just need to cast it as a List<TimeRange> first, and change the type to List<TimeRange> instead of ArrayList<>

Reply:
Thanks! It works now.